### PR TITLE
Refactor validation tools to supply more information

### PR DESCRIPTION
**Pull request by ZacBlanco**
I updated the tool interface so that it's easier to determine the list of possible tasks that can come from each tool, as well as run a specific task

### CODE REVIEW

File: integration/tools/hms/src/main/java/alluxio/cli/hms/MetastoreValidationTask.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import alluxio.collections.Pair;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+
+/**
+ * A validation task which includes extra information in order to provide dependencies for
+ * sub-tasks along with the task's validation result.
+ *
+ * @param <K> The input type of this type
```
Comment (LuQQiu) [inline integration/tools/hms/src/main/java/alluxio/cli/hms/MetastoreValidationTask.java:25]: type -> task

**Review by LuQQiu — APPROVED** [review]
LGTM for the hms related changes

**Review by jiacheliu3 — APPROVED** [review]
LGTM. Thanks for the refactor!

File: integration/tools/hms/src/main/java/alluxio/cli/hms/MetastoreValidationTask.java (hunk 1 above)
Comment (ZacBlanco) [reply integration/tools/hms/src/main/java/alluxio/cli/hms/MetastoreValidationTask.java]: updated

**Comment by ZacBlanco** [comment]
alluxio-bot, merge this please

**Pull request merged**